qiskit/algorithms/time_evolution/evolution_result.py

```diff
@@ -0,0 +1,39 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Class for holding evolution result and relevant metadata."""
+from typing import Dict, Any, Optional, Union
+
+from qiskit.opflow import OperatorBase
+
+
+class EvolutionResult:
+    """Class for holding evolution result and relevant metadata."""
+
+    def __init__(self, evolved_object: OperatorBase, metadata: Optional[Dict[str, Any]] = None):
+        """
+        Args:
+            evolved_object: An evolved quantum state or an evolved quantum observable.
+            metadata: A dictionary with algorithm-specific metadata. Keys contain strings that name
+                data stores as a corresponding value.
+        """
+        self._evolved_object = evolved_object
+        self._metadata = metadata
+
+    @property
+    def evolved_object(self) -> OperatorBase:
+        """Returns an evolved quantum state or an evolved quantum observable."""
+        return self._evolved_object
+
+    @property
+    def metadata(self) -> Union[Dict[str, Any], None]:
```

Comment:
I think it would be nice if we can specifically name attributes that certain algorithms should return, instead of having a general dictionary (to avoid general purposeddictionaries we introduced these result objects 🙂 ) . For example, a variational time evolution has an `ansatz`, so the variational algos should create a new result object that has an `ansatz` field. 

So I would think that we don't need the general `metadata` argument in this base class, but leave it for the derived classes to define more specific results 🙂

Reply:
Sounds reasonable. It would potentially generate the whole hierarchy of `EvolutionResult` children, though.